pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -202,8 +202,37 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus) *kubecontainer.Runtim
 	return &kubecontainer.RuntimeStatus{Conditions: conditions}
 }
 
+func (m *kubeGenericRuntimeManager) getSeccompProfile(annotations map[string]string, containerName string,
+	podSecContext *v1.PodSecurityContext, containerSecContext *v1.SecurityContext) string {
+	fieldProfile := func(scmp *v1.SeccompProfile) string {
+		if scmp.Type == v1.SeccompProfileUnconfined {
+			return v1.SeccompProfileNameUnconfined
+		}
+		if scmp.Type == v1.SeccompProfileTypeDefault {
+			return v1.SeccompProfileRuntimeDefault
+		}
+		if scmp.Type == v1.SeccompProfileLocalhost && scmp.LocalhostProfile != nil {
+			fname := filepath.Join(m.seccompProfileRoot, *scmp.LocalhostProfile)
+			return v1.SeccompLocalhostProfileNamePrefix + fname
+		}
+
+		return ""
+	}
+
+	if containerSecContext != nil && containerSecContext.SeccompProfile != nil {
+		return fieldProfile(containerSecContext.SeccompProfile)
+	}
+
+	if podSecContext != nil && podSecContext.SeccompProfile != nil {
+		return fieldProfile(podSecContext.SeccompProfile)
+	}
+
+	return m.getSeccompProfileFromAnnotations(annotations, containerName)
+}
+
 // getSeccompProfileFromAnnotations gets seccomp profile from annotations.
 // It gets pod's profile if containerName is empty.
+// Seccomp Annotations is deprecated and should be replaced by getSeccompProfile.
 func (m *kubeGenericRuntimeManager) getSeccompProfileFromAnnotations(annotations map[string]string, containerName string) string {
```

Comment:
I would suggest a single method that looks at container/pod fields and annotations, otherwise this is likely to attract call sites that don't honor the fields

Reply:
Done, I removed the older method.